doc/source/whatsnew/v0.21.0.txt

```diff
@@ -126,7 +126,7 @@ Other Enhancements
 - :func:`date_range` now accepts 'Y' in addition to 'A' as an alias for end of year (:issue:`9313`)
 - Integration with `Apache Parquet <https://parquet.apache.org/>`__, including a new top-level :func:`read_parquet` and :func:`DataFrame.to_parquet` method, see :ref:`here <io.parquet>`.
 - :func:`DataFrame.add_prefix` and :func:`DataFrame.add_suffix` now accept strings containing the '%' character. (:issue:`17151`)
-- `read_*` methods can now infer compression from non-string paths, such as ``pathlib.Path`` objects (:issue:`17206`).
+- Certain read/write methods can now infer compression from non-string paths, such as ``pathlib.Path`` objects (:issue:`17206`). Only io methods that rely on ``io.common._infer_compression`` are affected, which at this time is ``read_csv``, ``read_table``, ``read_pickle``, and ``write_pickle`` (:issue:`17262`).
```

Comment:
I don't think we should mention the private `io.common._infer_compression` in the whatsnew notes, I would just list the improved functions. (also write_pickle is not a public one -> to_pickle)